uiWidgets/CommandButton.go

```diff
@@ -41,20 +41,27 @@ func CreateCommandButton(
 }
 
 func (this *CommandButton) handleClicked() {
+	var handleClick func()
+	
 	if len(this.commandDefinition.Confirm) != 0 {
-		utils.MustConfirmDialogBox(this.parentWindow, this.commandDefinition.Confirm, this.sendCommand)
-		return
+		logger.Debugf("CommandButton.handleClicked().MustConfirmDialogBox: %s", this.commandDefinition.Name)
+		handleClick = utils.MustConfirmDialogBox(this.parentWindow, this.commandDefinition.Confirm, this.sendCommand)
```

Comment:
Seems like most of these changes aren't needed, and could have been fixed with just turning the call to MustConfirmDialogBox into an IIFE:
```
utils.MustConfirmDialogBox(this.parentWindow, this.commandDefinition.Confirm, this.sendCommand)()
```

I'm going to merge and close this PR and then clean up the changes here and in ControlButton up.

Reply:
You aren't wrong, that being said, I (personally) prefer single-return logic, and since calling of `handleClick()` is essentially the return on this function, I went this route.  But you aren't wrong.  Your call.